### ISSUE TITLE
oauth_providers_example.dart: example needs to be updated

### ISSUE THREAD

**Issue by diegonc** [nhost/nhost-dart#93]
The example's comments give wrong instructions regarding the configuration of NHost's OAuth callbacks:

At line [10](https://github.com/nhost/nhost-dart/blob/main/packages/nhost_flutter_auth/example/lib/oauth_providers_example.dart#L10) it reads:

```
/// Success redirect URL: `nhost-example://oauth.login.success`.
/// Failure redirect URL: `nhost-example://oauth.login.failure`.
```

wheareas only one callback URL can be set on GitHub settings and no URL can be set on NHost's GitHub provider configuration.

Also at line [23](https://github.com/nhost/nhost-dart/blob/main/packages/nhost_flutter_auth/example/lib/oauth_providers_example.dart#L23), the sign in URL is wrong. It should look like:
```
/// Fill in this value with the backend URL found on your Nhost project page.
const nhostGithubSignInUrl = '$nhostUrl/v1/auth/signin/provider/github';
```

**Comment by diegonc** [comment]
There also seems to be a mismatch between the expected query parameter name (`refresh_token`) and the actual parameter name used by NHost callback (`refreshToken`)

![image](https://user-images.githubusercontent.com/261458/204574871-39c8611f-7ec9-4e5c-b43a-c995bf06aa91.png)

**Comment by mhadaily** [comment]
Hi, 
I have updated all examples ready on this PR https://github.com/nhost/nhost-dart/pull/95

We are activity working to improve the SDK. stay tuned.

**Comment by diegonc** [comment]
Hi, thank you very much for your work!

Will this improvements include the migration from the old backend URL discussed in the link below?

https://github.com/nhost/nhost/discussions/1319

**Comment by mhadaily** [comment]
Yes, I am working on it. I think we will make it in the next major release two.

**Comment by mhadaily** [comment]
@diegonc  Check this out `nhost_sdk: ^4.0.0-dev.6` or [this link ](https://pub.dev/packages/nhost_sdk/versions/4.0.0-dev.6) for supporting subdomain and region.

**Comment by totzk9** [comment]
Unfortunately, using `launchUrl` with Facebook would block me from authenticating because of Embedded browser security check from Facebook.

Found a workaround to authenticate with OAuth in Flutter though.

1. Use this package https://pub.dev/packages/oauth_webauth specifically the [BaseWebView widget](https://pub.dev/packages/oauth_webauth#basewebview)
2. add  `initialUrl` with '${createNhostServiceEndpoint(subdomain: YOUR_SUBDOMAIN, region: YOUR_REGION, service: 'auth')}/signin/provider/facebook'
3. add `redirectUrls` with your callback URL (https://xxxxxx.auth.YOUR_REGION.nhost.run/v1/signin/provider/facebook/callback)

After successful auth, should return with the `callback?code=`

**Comment by dbarrosop** [comment]
> because it refers to success/failure callback settings

that part is incorrect, there is no success/failure URLs in hasura-auth, instead, if there is an error, the callback url will include an error query arg describing the issue
